samtranslator/model/api/http_api_generator.py

```diff
@@ -489,14 +489,21 @@ def _set_default_authorizer(self, open_api_editor, authorizers, default_authoriz
                 path, default_authorizer, authorizers=authorizers, api_authorizers=api_authorizers
             )
 
-    def _get_authorizers(self, authorizers_config, default_authorizer=None):
+    def _get_authorizers(self, authorizers_config, enable_iam_authorizer=False):
         """
         Returns all authorizers for an API as an ApiGatewayV2Authorizer object
         :param authorizers_config: authorizer configuration from the API Auth section
-        :param default_authorizer: name of the default authorizer
+        :param enable_iam_authorizer: if true add an "AWS_IAM" authorizer
         """
         authorizers = {}
 
+        if enable_iam_authorizer:
+            authorizers["AWS_IAM"] = ApiGatewayV2Authorizer(is_aws_iam_authorizer=True)
+
+        # If all the customer wants to do is enable the IAM authorizer the authorizers_config will be None.
+        if authorizers_config == None:
+            return authorizers
+
         if not isinstance(authorizers_config, dict):
             raise InvalidResourceException(self.logical_id, "Authorizers must be a dictionary.")
 
```

Comment:
Sorry not able to add a comment down below, but we should maybe add a condition in the `for` loop to ignore or fail on finding an `AWS_IAM` object in `authorizers_config.items()` so that we don't overwrite the one defined by `enable_iam_authorizer`

Reply:
I think that's fair. I was originally thinking we'd want to allow an overwrite but that was when this change was opt-out. Now that it is opt-in we shouldn't let users shoot themselves in the foot.

Reply:
Yea. if they already have an authorizer called "AWS_IAM" SAM doesn't clobber/overwrite it. The user's authorizer is used (though it likely won't help them).